polars/polars-arrow/src/kernels/rolling/no_nulls/mean.rs

```diff
@@ -35,41 +35,35 @@ pub fn rolling_mean<T>(
 where
     T: NativeType + Float + std::iter::Sum<T> + SubAssign + AddAssign + IsFloat,
 {
-    match (center, weights) {
-        (true, None) => rolling_apply_agg_window::<MeanWindow<_>, _, _>(
+    let offset_fn = match center {
+        true => det_offsets_center,
+        false => det_offsets,
+    };
+    match weights {
+        None => rolling_apply_agg_window::<MeanWindow<_>, _, _>(
             values,
             window_size,
             min_periods,
-            det_offsets_center,
+            offset_fn,
             None,
         ),
-        (false, None) => rolling_apply_agg_window::<MeanWindow<_>, _, _>(
-            values,
-            window_size,
-            min_periods,
-            det_offsets,
-            None,
-        ),
-        (true, Some(weights)) => {
-            let weights = no_nulls::coerce_weights(weights);
-            no_nulls::rolling_apply_weights(
-                values,
-                window_size,
-                min_periods,
-                det_offsets_center,
-                no_nulls::compute_mean_weights,
-                &weights,
-            )
-        }
-        (false, Some(weights)) => {
-            let weights = no_nulls::coerce_weights(weights);
+        Some(weights) => {
+            // A weighted mean is a weighted sum with normalized weights
+            let mut wts = no_nulls::coerce_weights(weights);
+            let wsum = wts.iter().fold(T::zero(), |acc, x| acc + *x);
+            assert_ne!(
```

Comment:
We should return an error, or maybe do the weights validation a level higher in the function that does returns a `PolarsResult`.

Reply:
This function gets called from several higher level functions so I'd have to replicate the check and I'm not nuts about that.
Would returning an error require changing the function signature here? If so we'd have to change all the rolling kernel signatures I think.

Reply:
> Would returning an error require changing the function signature here? If so we'd have to change all the rolling kernel signatures I think.

That's also fine. We can wrap the results in an `Ok`.